library/src/main/scala/scala/scalajs/concurrent/QueueExecutionContext.scala

```diff
@@ -6,13 +6,17 @@ import scalajs.js
 private[concurrent] object QueueExecutionContext
     extends ExecutionContextExecutor {
 
-  def execute(runnable: Runnable) = {
-    val lambda: js.Function = () =>
-      try { runnable.run() } catch { case t: Throwable => reportFailure(t) }
-    js.Dynamic.global.setTimeout(lambda, 0)
+  def execute(runnable: Runnable): Unit = {
+    js.timers.setTimeout(0) {
+      try {
+        runnable.run()
+      } catch {
+        case t: Throwable => reportFailure(t)
+      }
+    }
   }
 
-  def reportFailure(t: Throwable) =
-    Console.err.println("Failure in async execution: " + t)
+  def reportFailure(t: Throwable): Unit =
+    t.printStackTrace()
```

Comment:
I think it might still be valuable to have a message that says this was an asynchronous computation.